### PR TITLE
Bjorncs/jdisc metric unhandled exception

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

File: container-core/src/main/java/com/yahoo/container/jdisc/ThreadedRequestHandler.java [hunk 1]
```diff
@@ -97,9 +97,12 @@ protected ThreadedRequestHandler(Executor executor, Metric metric, boolean allow
         dimensions.put("port", Integer.toString(uri.getPort()));
         String handlerClassName = getClass().getName();
         dimensions.put("handler-name", handlerClassName);
+        dimensions.putAll(extraDimensions);
         return this.metric.createContext(dimensions);
     }
 
+    private Metric.Context contextFor(Request request) { return contextFor(request, Map.of()); }
```
Comment (gjoranv) [inline container-core/src/main/java/com/yahoo/container/jdisc/ThreadedRequestHandler.java]: What is the difference between Map.of() and Collections.emptyMap()?

File: container-core/src/main/java/com/yahoo/container/jdisc/ThreadedHttpRequestHandler.java [hunk 2]
```diff
@@ -31,6 +32,7 @@
 
     public static final String CONTENT_TYPE = "Content-Type";
     private static final String RENDERING_ERRORS = "rendering_errors";
+    private static final String UNHANDLED_EXCEPTION_METRIC = "jdisc.http.handler.unhandled_exception";
```
Comment (gjoranv) [inline container-core/src/main/java/com/yahoo/container/jdisc/ThreadedHttpRequestHandler.java]: Shouldn't it be `..unhandled_exceptions` with the plural 's'? If so, also rename the constant.

**Review by gjoranv — COMMENTED** [review]
Other counter metrics use a plural 's' so I suggest to do that, and rename everywhere, constant, test method etc.

File: container-core/src/main/java/com/yahoo/container/jdisc/ThreadedRequestHandler.java (hunk 1 above)
Comment (bjorncs) [reply container-core/src/main/java/com/yahoo/container/jdisc/ThreadedRequestHandler.java]: `Map.of()` was introduced in Java 9. The returned `Map` instance from both have the same semantics (immutable etc). I prefer the new syntax as it is shorter.

File: container-core/src/main/java/com/yahoo/container/jdisc/ThreadedHttpRequestHandler.java (hunk 2 above)
Comment (bjorncs) [reply container-core/src/main/java/com/yahoo/container/jdisc/ThreadedHttpRequestHandler.java]: Agree - plural sounds better. I will update the PR with a rename.

File: container-core/src/main/java/com/yahoo/container/jdisc/ThreadedRequestHandler.java (hunk 1 above)
Comment (gjoranv) [reply container-core/src/main/java/com/yahoo/container/jdisc/ThreadedRequestHandler.java]: When you static import the `emptyMap()` function, I think that is more describing and almost as short. But this is just nit of course. :)

File: container-core/src/main/java/com/yahoo/container/jdisc/ThreadedHttpRequestHandler.java (hunk 2 above)
Comment (gjoranv) [reply container-core/src/main/java/com/yahoo/container/jdisc/ThreadedHttpRequestHandler.java]: 👍

File: container-core/src/main/java/com/yahoo/container/jdisc/ThreadedRequestHandler.java (hunk 1 above)
Comment (bjorncs) [reply container-core/src/main/java/com/yahoo/container/jdisc/ThreadedRequestHandler.java]: Agree - if static import is used `emptyMap()` is more readable 👍

**Comment by bjorncs** [comment]
@gjoranv PTAL

**Review by gjoranv — APPROVED** [review]
Optional: I suggest to rename the test method and also the metric constants to reflect that the metric names are 'exceptions' and not 'exception'.

**Review by gjoranv — APPROVED** [review]

**Pull request merged**